apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/controller/ConsumerController.java

```diff
@@ -79,10 +79,20 @@ public ConsumerToken getConsumerTokenByAppId(@RequestParam String appId) {
 
   @PreAuthorize(value = "@permissionValidator.isSuperAdmin()")
   @PostMapping(value = "/consumers/{token}/assign-role")
-  public List<ConsumerRole> assignNamespaceRoleToConsumer(@PathVariable String token,
-                                                          @RequestParam String type,
-                                                          @RequestParam(required = false) String envs,
-                                                          @RequestBody NamespaceDTO namespace) {
+  public List<ConsumerRole> assignNamespaceRoleToConsumer(
+      @PathVariable String token,
+      @RequestParam(required = false, defaultValue = "false") boolean isAllowCreateApplication,
+      @RequestParam String type,
+      @RequestParam(required = false) String envs,
+      @RequestBody NamespaceDTO namespace
+  ) {
+    List<ConsumerRole> consumerRoleList = new ArrayList<>(8);
+    if (isAllowCreateApplication) {
```

Comment:
Shall we also update the open platform authorization page to include the "allow create application" feature?

Reply:
The UI updated.

Now, only allowCreateApplication when create then consumer, when consumer created, user cannot change it.

![image](https://github.com/apolloconfig/apollo/assets/15523186/759c4c07-666d-4b96-b6f9-5cce3c1ea25b)

And user can check it by the page list of consumer

![image](https://github.com/apolloconfig/apollo/assets/15523186/9de2c7a4-3faf-43b4-9d77-915d828f39ae)

Reply:
The button(preGrantPermission) on the right don't work after add this column... 

It seems that a bug occour

Reply:
Fixed in commit https://github.com/apolloconfig/apollo/pull/4954/commits/19ce123ee5e72f5c97b4cf59cf172833361be11c

----

Sometime angularjs cannot translate the placehoder

![image](https://github.com/apolloconfig/apollo/assets/15523186/d948a04d-5072-4f08-b7c1-f6c791a2a290)

Is there a way to fix it?

Reply:
Is it a cache problem?

Reply:
It's a cache problem.

Handle it by press `Ctrl + F5` already